streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1805,12 +1816,12 @@ private void completeShutdown(final boolean cleanRun) {
             log.error("Failed to unsubscribe due to the following error: ", e);
         }
         try {
-            mainConsumer.close();
+            mainConsumer.close(closeOptions);
         } catch (final Throwable e) {
             log.error("Failed to close consumer due to the following error:", e);
         }
         try {
-            restoreConsumer.close();
+            restoreConsumer.close(closeOptions);
```

Comment:
the restore consumer direct assignment so it's not part of the consumer group and doesn't need the close options (I don't remember, what happens if you pass in LEAVE_GROUP or REMAIN_IN_GROUP for a consumer that doesn't use group subscription?)

Reply:
If the consumer doesn't subscribe, it will not join a group.
In this case, it will get an unknown member ID in the response.

Reply:
I think we should pass either DEFAULT or REMAIN_IN_GROUP in for the restore consumer so it doesn't try to leave the group. If I'm remembering correctly, when LEAVE_GROUP is used it always tries to send a leave group even if it's not in a group, whereas with DEFAULT it will skip the leaveGroup if there's no member id or the coordinator is unknown

Reply:
Hm actually how could it even send a LeaveGroup if LEAVE_GROUP is selected for a consumer that isn't part of a group? That is, by definition the coordinator is unknown, so where would the request even be sent?

Don't remember of the top of my head how we handled this in the consumer client CloseOptions PR but it seems like the consumer should be agnostic to the GroupMembershipOperation and never send anything if it's not part of a group to begin with, right?